video/agon.h

```diff
@@ -86,6 +87,9 @@
 #define AUDIO_CMD_ENABLE		8		// Enables a channel
 #define AUDIO_CMD_DISABLE		9		// Disables (destroys) a channel
 #define AUDIO_CMD_RESET			10		// Reset audio channel
+#define AUDIO_CMD_SEEK			11		// Seek to a position in a sample
+#define AUDIO_CMD_DURATION		12		// Set the duration of a channel
+#define AUDIO_CMD_SAMPLERATE	13		// Set the samplerate for channel or underlying audio system

```

Comment:
new commands!

Seek will only work on channels set up for sample playback - it's kinda meaningless for other waveform types

Duration allows for note playback duration to be changed.  this will kick off a new playback if the channel is not already playing a note.  volume will be at the last volume used on the channel, which now defaults to 64 (i.e. 50%)

Samplerate allows for the setting of the sample rate for an individual channel, or if the channel for the command is set to -1 it will set the rate for the underlying audio system.  When setting the rate for the underlying system, the rate on all individual channels is changed to match.  Changing the rate on an individual channel "tunes" that channel relative to the underlying audio system - if it's set to a higher rate than the underlying system the channel doesn't actually result in more samples being produced for that channel.

Changing the sample rate on the underlying audio system increases the fidelity of the audio system at the cost of some processing.  The pitch/frequency of notes (and samples) played after changing the sample rate shouldn't be affected, and will still play the same notes as before.

(there is actually a bug with the note playback frequency of non-sample channels - if you have played a note at frequency X on a channel, change the underlying system sample rate , and then play another note on the same channel also at frequency X then it will be pitch shifted.  the work-around/fix to this is to play a note at a different frequency first.  this also affects notes that may be playing when the underlying sample rate is changed)